Make cursor not-allowed when hovering groups where task cannot be dro…

PatricKangasmaki opened the pull request:
Cursor is now type "not-allowed" when hovering task groups that aren't allowed

juhoeerola commented on frontend/src/components/TaskMapTask.module.scss:
```diff
@@ -48,6 +48,10 @@
   &.loading:not(.dragging, .draggingOutside) {
     cursor: wait;
   }
```
Looking at this, the loading cursor style should also be applied to all the cases where the `not-allowed` is (the group, and the handles), as it now doesn't seem to show consistently.

juhoeerola reviewed (approved):
Looking good 👍 

The loading cursor might be good to fix now at the same time, as it is needed in all the same places. Easier now than hunting those down later.

juhoeerola commented on frontend/src/components/TaskMapEdge.tsx:
```diff
@@ -8,7 +8,14 @@ import { apiV2 } from '../api/api';
 const classes = classNames.bind(css);
 
-const DrawPath: FC<any> = ({ id, d, markerEnd, disableInteraction }) => {
+const DrawPath: FC<any> = ({
```
Some types might be good to add for this (as well as `CustomEdge`'s props)

juhoeerola commented on frontend/src/components/TaskMapTask.tsx:
```diff
@@ -26,7 +26,10 @@ export interface TaskProps {
   checked: { from: boolean; to: boolean };
   disableDragging: boolean;
+  dropDisabled?: boolean;
+  setGroupDraggable: any;
```
```suggestion
  setGroupDraggable: (_: boolean) => unknown;
```

Could be a suitable type, since that is what it is used for

juhoeerola commented on frontend/src/components/TaskMapTask.tsx:
```diff
@@ -140,8 +157,18 @@ const SingleTask: FC<
+  taskId,
+  index,
+  disableDragging,
+  dropDisabled,
+  setGroupDraggable,
+  draggingSomething,
+  isLoading,
+  ...rest
```
```suggestion
  ...rest
```

These could be included in the rest, as these are only forwarded

juhoeerola commented on frontend/src/components/TaskMapTask.tsx:
```diff
@@ -155,6 +182,10 @@ export const DraggableSingleTask: FC<
           provided={provided}
           snapshot={snapshot}
           disableDragging={disableDragging}
+          dropDisabled={dropDisabled}
+          setGroupDraggable={setGroupDraggable}
+          draggingSomething={draggingSomething}
+          isLoading={isLoading}
           {...rest}
```
```suggestion
          {...rest}
```

juhoeerola commented on frontend/src/components/TaskMapTask.tsx:
```diff
@@ -140,8 +157,18 @@ const SingleTask: FC<
 export const DraggableSingleTask: FC<
   TaskProps & {
     index: number;
+    draggingSomething: boolean;
```
This might fit better in `TaskProps`, since it is always used with that type

PatricKangasmaki commented:
Great points, added all your suggestions 👍 I also fixed the bug where not-allowed cursor didn't show up correctly when hovering over a single task (the bug which showed up in the demo). Css was just missing one class from the :not pseudo class

`&.draggingSomething:not(.loading) { ...` to
`&.draggingSomething:not(.loading, .dropDisabled) { ...`

juhoeerola reviewed (approved):
👍 seems to work fine (also on safari)
